Core/Model/User.php

```diff
@@ -295,26 +295,28 @@ protected function saveInsert(array $values = [])
     protected function setNewRole()
     {
         $roleModel = new Role();
-        foreach ($roleModel->all() as $role) {
-            $roleUser = new RoleUser();
-            $roleUser->codrole = $role->codrole;
-            $roleUser->nick = $this->nick;
-            $roleUser->save();
-
-            /// set user homepage
-            foreach ($roleUser->getRoleAccess() as $roleAccess) {
-                $this->homepage = $roleAccess->pagename;
-                if ('List' == \substr($this->homepage, 0, 4)) {
-                    break;
-                }
+        $role = $roleModel->defaultRole();
```

Comment:
Por favor, haz la búsqueda del rol por defecto aquí y no en el modelo role.

Reply:
The change is already applied.